### DISCUSSION TITLE
How to login with API?

### DISCUSSION THREAD

**Discussion by seanmars** [directus/directus#3283] [Archive]
In document [api/authentication](https://docs.directus.io/reference/api/authentication.html).
I can not found any information of query(parameter) for login(and many API).

**Comment by rijkvanzanten** [comment]
You send a POST request to `/auth/login` with the following payload:

```json
{
  "email": "admin@example.com",
  "password": "password"
}
```

**Reply by benhaynes** [reply]
Ref: https://github.com/directus/directus/issues/3284

**Comment by jakobgreenfeld** [comment]
I'm currently having the same problem.  I tried sending a POST request with a payload containing my email and password to http://directus.mydomain.com/auth/login. 

But this yields the error 

`{'error': {'code': 3, 'message': 'Unauthorized request', 'class': 'Directus\Exception\UnauthorizedException', 'file': '/app/code/src/helpers/app.php', 'line': 287}}`

Since my login details are correct, I'm assuming that I use the wrong endpoint.  I searched everywhere and people seem to be using endpoints like http://directus.mydomain.com/api/auth/login or http://directus.mydomain.com/api/1.1/auth/login or http://directus.mydomain.com/api/auth/authenticate. None of them worked either and I suppose they were used in previous versions of directus. 

My code 

```
payload = {
    'email': 'jakob@mydomain.com',
    'password': 'mypassword'
}

req = requests.post('https://directus.mydomain.com/auth/login', data =payload)
```

**Reply by jakobgreenfeld** [reply]
I figured it out. The correct endpoint reads https://directus.mydomain.com/project/auth/authenticate As far as I can see this is not documented which seems rather strange given that this is one of the most basic operations.

**Reply by benhaynes** [reply]
Ref: #3284

**Reply by philleepflorence** [reply]
@benhaynes  - Is this required for the docs?

**Reply by benhaynes** [reply]
Haha, everything should be in the Docs... so yup! 👍

**Reply by panchmp** [reply]
```
curl -X POST -k -i 'https://directus.somedomain.com/auth/login' --data '{"email":"admin@example.com","password":"password","mode":"cookie"}'
```
returns 400 error code
```
{"errors":[{"message":"\"email\" is required","extensions":{"code":"INVALID_PAYLOAD"}}]}
```
@rijkvanzanten What I am doing wrong?

**Reply by panchmp** [reply]
Correct example
```
curl -X POST -H 'Accept: application/json, text/plain, */*' -H 'Content-Type: application/json;charset=UTF-8' -i http://localhost:8055/auth/login --data '{"email":"admin@example.com","password":"password","mode":"cookie"}'
```

**Comment by rijkvanzanten** [comment]
See the updated docs 🙂 

https://docs.directus.io/reference/authentication/#login

**Comment by panchmp** [comment]
> Hello, do you remember how you
> solved this? I am facing the same issue and can't seem to find a solution
> for it. i tried calling the endpoint in all sort of different ways, but I
> always get the same error as you did.
> Any help will be appreciated,
> Thanks!

    curl -X POST -H 'Accept: application/json, text/plain, */*' -H
    'Content-Type: application/json;charset=UTF-8' -i
    http://localhost:8055/auth/login --data
    '{"email":"***@***.***","password":"password","mode":"cookie"}'
